tests/object/test_object_spec.lua

```diff
@@ -0,0 +1,120 @@
+-- Luagit2's Object Module Tests
+
+local fixer = require("Fixtures.fix_repo")
+
+describe(" Object Methods Tests ", function()
+	local lib = require("luagit2")
+	local repo
+	local blob_id_str = "1385f264afb75a56a5bec74243be9b367ba4ca08"
+	local tree_id_str = "181037049a54a1eb5fab404658a3a250b44335d7"
+	local tag_id_str = "0c37a5391bbff43c37f0d0371823a5509eed5b1d"
+	local commit_id_str = "108ddee361877aa5c044d89d8dd232b8fd0f8992"
+	local blob_id, tree_id, tag_id, commit_id
+	local otype_blob, otype_tree, otype_tag, otype_commit
+	local obj_blob, obj_tree, obj_tag, obj_commit
+	local current_directory_path = lfs.currentdir()
+
+	setup(function()
+		lib.luagit2_init()
+	end)
+
+	before_each(function()
+		fixer.set_repo("new_test_repo")
+		repo = lib.luagit2_repository_open("Fixtures/WORKON_REPO/.git")
+		blob_id = lib.luagit2_oid_fromstr(blob_id_str)
+		tree_id = lib.luagit2_oid_fromstr(tree_id_str)
+		tag_id = lib.luagit2_oid_fromstr(tag_id_str)
+		commit_id = lib.luagit2_oid_fromstr(commit_id_str)
+		otype_blob = lib.luagit2_object_string2type("blob")
+		otype_tree = lib.luagit2_object_string2type("tree")
+		otype_tag = lib.luagit2_object_string2type("tag")
+		otype_commit = lib.luagit2_object_string2type("commit")
+		obj_blob = lib.luagit2_object_lookup(repo,blob_id,otype_blob)
+		obj_tree = lib.luagit2_object_lookup(repo,tree_id,otype_tree)
+		obj_tag = lib.luagit2_object_lookup(repo,tag_id,otype_tag)
+		obj_commit = lib.luagit2_object_lookup(repo,commit_id,otype_commit)
+	end)
+
+	after_each(function()
+		lib.luagit2_object_free(obj_blob)
+		lib.luagit2_object_free(obj_tree)
+		lib.luagit2_object_free(obj_tag)
+		lib.luagit2_object_free(obj_commit)
+		lib.luagit2_repository_free(repo)
+		fixer.set_back()
+	end)
+
+	teardown(function()
+		lib.luagit2_shutdown()
+	end)
+
+	--[[ A part of the ` git cat-file --batch-check --batch-all-objects `
+		 on tests/new_test_repo
+
+		 for tests we have used these values
+
+		0c37a5391bbff43c37f0d0371823a5509eed5b1d tag 144         | (for obj_tag)
+		108ddee361877aa5c044d89d8dd232b8fd0f8992 commit 236      | (for obj_commit)
+		1385f264afb75a56a5bec74243be9b367ba4ca08 blob 4          | (for obj_blob)
+		181037049a54a1eb5fab404658a3a250b44335d7 tree 34         | (for obj_tree)
+		1810dff58d8a660512d4832e740f692884338ccd tree 112
+		1b49a09ae5e707b8ed48e6048aa1901138838944 tree 137
+		2d2eff63372b08adf0a9eb84109ccf7d19e2f3a2 commit 163
+		36060c58702ed4c2a40832c51758d5344201d89a commit 227
+		418382dff1ffb8bdfba833f4d8bbcde58b1e7f47 tree 39
+		41bc8c69075bbdb46c5c6f0566cc8cc5b46e8bd9 commit 230
+		45b983be36b73c0788dc9cbcb76cbb80fc7bb057 blob 3
+		4a202b346bb0fb0db7eff3cffeb3c70babbd2045 commit 227
+		5001298e0c09ad9c34e4249bc5801c75e9754fa5 commit 182
+		5b5b025afb0b4c913b4c338a42934a3863bf3644 commit 227
+
+	]]--
+
+	it("Tests getting object ID ",function()
+		local check_blob_id = lib.luagit2_object_id(obj_blob)
+		local check_blob_id_str = lib.luagit2_oid_tostr_s(check_blob_id)
```

Comment:
I dont think you needed to test all these different cases, because it's not like we are testing libgit2 itself, just the lua connection :)